### PR TITLE
Removed unittests stage from workflow

### PR DESCRIPTION
**Pull request by rbudhrani**

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -73,8 +73,8 @@ jobs:
         run: |
           pip install mypy
           mypy --namespace-packages $SOURCE_DIRS | tee mypy.log
-          if [ -n "$( tail -n 1 mypy.log | grep -e '^Succes' )" ]; then RESULT="passed"; else RESULT="failed"; fi
-          if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l mypy -v $RESULT -f $BADGES_DIR/mypy.svg failed=red passed=green; fi
+          if [ -n "$( tail -n 1 mypy.log | grep -e '^Succes' )" ]; then RESULT="passing"; else RESULT="failing"; fi
+          if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l mypy -v $RESULT -f $BADGES_DIR/mypy.svg failing=red passing=green; fi
```
Comment (heevasti) [inline .github/workflows/pull_request.yml:77]: The mypy tests either "passed" or "failed". They are not "passing" or "failing" at any moment. ;) Please change this back.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -115,33 +115,24 @@ jobs:
       - name: coverage
         run: |
           pip install coverage
-          chmod a+x scripts/run_unit_tests_with_coverage.sh
-          scripts/run_unit_tests_with_coverage.sh $SOURCE_DIRS
+          if coverage run --branch -m unittest discover --start-directory=tests --pattern="test_*.py"; then RESULT="passing"; else RESULT="failing"; fi
           coverage report --show-missing --fail-under=$COVERAGE_MIN_PERC | tee coverage.log
           COVERAGE_PERC=$(tail -1 coverage.log | grep -o '....$')
           if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l coverage -v $COVERAGE_PERC -f $BADGES_DIR/coverage.svg 60=red 80=orange 100=green; fi
+          if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l tests -v $RESULT -f $BADGES_DIR/tests.svg failing=red passing=green; fi
```
Comment (heevasti) [inline .github/workflows/pull_request.yml:122]: Please change to "passed" and "failed".

File: .github/workflows/github-ci.yml [hunk 3]
```diff
@@ -60,7 +60,7 @@ jobs:
         run: |
           pip install mypy
           mypy --namespace-packages $SOURCE_DIRS | tee mypy.log
-          if [ -n "$( tail -n 1 mypy.log | grep -e '^Succes' )" ]; then RESULT="passed"; else RESULT="failed"; fi
+          if [ -n "$( tail -n 1 mypy.log | grep -e '^Succes' )" ]; then RESULT="passing"; else RESULT="faling"; fi
```
Comment (heevasti) [inline .github/workflows/github-ci.yml:63]: Please change back to "passed" and "failed".

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (rbudhrani) [reply .github/workflows/pull_request.yml:77]: the common convention it seems is to use "passing" and "failing" instead of the past tense

Comment (rbudhrani) [reply .github/workflows/pull_request.yml:77]: As discussed within the team, we now choose to go our own way with "pass" and "fail"

**Review by heevasti — DISMISSED** [review]

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -115,33 +115,24 @@ jobs:
       - name: coverage
         run: |
           pip install coverage
-          chmod a+x scripts/run_unit_tests_with_coverage.sh
-          scripts/run_unit_tests_with_coverage.sh $SOURCE_DIRS
+          if coverage run --branch -m unittest discover --start-directory=tests --pattern="test_*.py"; then RESULT="pass"; else RESULT="fail"; fi
           coverage report --show-missing --fail-under=$COVERAGE_MIN_PERC | tee coverage.log
           COVERAGE_PERC=$(tail -1 coverage.log | grep -o '....$')
           if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l coverage -v $COVERAGE_PERC -f $BADGES_DIR/coverage.svg 60=red 80=orange 100=green; fi
+          if [ "${{ matrix.python-version }}" = "3.10" ]; then anybadge -o -l tests -v $RESULT -f $BADGES_DIR/tests.svg fail=red pass=green; fi
```
[the remark this answers is not in the document]
Comment (rbudhrani) [reply .github/workflows/pull_request.yml]: yep i missed the `--source` option, which is now back in

**Review by heevasti — APPROVED** [review]

**Pull request merged**